prober/http.go

```diff
@@ -550,7 +550,7 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 
 		if !requestErrored {
 			_, err = io.Copy(io.Discard, byteCounter)
-			if err != nil {
+			if err != nil && (errors.Is(err, &http.MaxBytesError{}) || httpConfig.GetFailIfBodyTooLarge()) {
```

Comment:
Please check if this is okay or if I should move this to another `if` so it's clearer.

Reply:
I would prefer another if for clarity